tests/test_convert.py

```diff
@@ -558,8 +558,23 @@ def test_gate_broadcast():
     expected.cx(q, p)
     expected.cx(q, p[0])
     assert qc == expected
-    # We need to be safe in the event of mutation.
-    assert len({id(x.operation) for x in qc.data}) == len(qc.data)
+    # We need to be safe in the event of mutation.  We can be safe either if the object is
+    # immutable, or if it's mutable but only present once in the circuit.
+    num_safe_mutate = 0
+    unsafe_mutate_indices = []
```

Comment:
is there any particular reason why this is a list and not a bool? wondering because all you do with it is  `assert not unsafe_mutate_indices`

Reply:
The idea is that if there's a test failure, the failure message will show all the invalid locations, making it easier to debug.  The test runner in this repo is `pytest`, which shows the values of the inner objects on `assert` lines, so it'll show both the `False` from `not [...]` and it'll show the list itself.